sdk/src/rest/types.ts

```diff
@@ -0,0 +1,13 @@
+export interface RestCommand<_Input extends object, _Output extends object | unknown, _Schema extends object> {
+	(): RestRequestOptions;
+}
+
+export type HttpMethod = 'GET' | 'SEARCH' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+
+export interface RestRequestOptions {
+	method: HttpMethod;
+	path: string;
+	params?: Record<string, string>;
```

Comment:
Standardizing this to the `URLSearchParams` input makes setting those on the URL object a little more reliable